### ISSUE TITLE
simulate('click') on child component does not propagate to trigger blur on parent

### ISSUE THREAD

**Issue by jamesopti** [airbnb/enzyme#1551]
#### API

- [ X ] shallow

#### Version

| library         | version
| ---------------- | -------
| Enzyme           |     3.0
| React            |    16.0

#### Adapter

- [ X ] enzyme-adapter-react-16

I have a simple dropdown component that I'm having trouble regression testing.
In Dropdown.js below, a bug was introduced which broke children's click handlers.

The onBlur event on the parent fires before the click event in the child, despite being triggered by the child click. This resulted in the child being destroyed and its clicks not being registered.
(the bug was removing the onMouseOver/onMouseLeave handlers, which set state that onBlur uses)

Enzyme tests didn't catch this because they triggered a click directly in one
of the children, circumventing the event propagation of the blur.

Is there a better way to test this using enzyme?

Dropdown.js - psuedocode
```
return (
  <div>
    <button
      onClick={ this.toggleOpenClose }
      onBlur={ this.setIsOpenFalse }>
      <div className='flex'>
        { buttonContent }
      </div>
    </button>
    <div
      onMouseOver={ this.handleMouseOverChildren }
      onMouseLeave={ this.handleMouseLeavingChildren }
      onClick={ this.setIsOpenFalse } >
      { this.state.isOpen && children }
    </div>
  </div>
);
```

test.js

```
it('should call the child onClick handler when clicked', () => {
  const onClickMock = jest.fn();
  const component = shallow(
    <Dropdown>
      <ul>
        {
          data.map((item, index) => {
            return (
              <li
                key={ index }
                onClick={ onClickMock }
                data-test-section={ `dropdown-item-${index}` }>
                { item.title }
              </li>
            );
          })
        }
      </ul>
    </Dropdown>
  );

  component.find('[data-test-section="dropdown-item-1"]').simulate('click');
  expect(onClickMock.mock.calls.length).toBe(1);
});
```

**Comment by ljharb** [comment]
Because you've provided pseudocode, I can't be sure - can you provide the real code?

**Comment by jamesopti** [comment]
@ljharb sure thing! Just updated with minimally reproducible example.

**Comment by jamesopti** [comment]
@ljharb Any thoughts after seeing the real code?

**Comment by ljharb** [comment]
@jamesopti yes. one problem is that you're using arrow functions in class properties, which you should never do for many reasons. Specifically here, you can't mock them out in tests (altho this might be a red herring).

Separately, I'd suggest avoiding `simulate` entirely. If you want to invoke the onClick prop, use `.prop('onClick')()`.
